lib/components/eureka_camera_form.dart

```diff
@@ -0,0 +1,250 @@
+    }
+  }
+
+  /// Using image_picker package, you will need to pass in the source of
+  /// the image (camera or gallery)
+  Future<void> _getImage(ImageSource source) async {
+    try {
+      _image = await widget.picker.getImage(
+        source: source,
+        imageQuality: 25,
+      );
+    } catch (e) {
+      print(e);
+    }
+
+    setState(() {
+      widget.mediaPaths.add(_image.path);
+    });
+  }
+
+  /// Camera buttons that lets users add new images, will resize to fit
+  /// the proper form
+  FlatButton cameraFormButtons({
+    @required Function onPressed,
+    @required IconData icon,
+    @required String text,
+  }) {
+    bool isEmpty = widget.mediaPaths.length > 0;
+
+    return FlatButton(
+      onPressed: onPressed,
+      child: Padding(
+        padding:
+            isEmpty ? const EdgeInsets.all(0.0) : const EdgeInsets.all(16.0),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Icon(icon),
+            SizedBox(
+              width: isEmpty ? 5.0 : 10.0,
+            ),
+            Text(
+              text,
+              style: TextStyle(
+                fontSize: isEmpty ? 12.0 : 16.0,
+                fontWeight: isEmpty ? FontWeight.normal : FontWeight.bold,
+              ),
+            ),
+          ],
+        ),
+      ),
+      color: Colors.grey[200],
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(20.0),
+      ),
+    );
+  }
+
+  /// These are the smaller buttons that will show along side a grid of image
+  /// preview. This will only show if the user already have taken a picture.
+  Row _filledMediaPathsAddImageButtons() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      children: [
+        cameraFormButtons(
+          onPressed: () async => _getImage(ImageSource.camera),
+          icon: Icons.camera_alt,
+          text: 'Take a New Photo',
+        ),
+        cameraFormButtons(
+          onPressed: () async => _getImage(ImageSource.gallery),
+          icon: Icons.photo_library,
+          text: 'Pick Photo from Gallery',
+        ),
+      ],
+    );
+  }
+
+  /// This allows users to open an image fullscreen to preview
+  /// and zoom their image.
+  void _fullScreenImagePreview(int index) {
+    Navigator.push(
+      context,
+      MaterialPageRoute(
+        builder: (context) =>
+            EurekaImageViewer(imagePath: widget.mediaPaths[index]),
+      ),
+    );
+  }
+
+  /// These are the smaller image previews in the grid, you can tap on the
+  /// image to expand into a full screen view.
+  Container _showImagePreview(int index) {
+    return Container(
+      width: double.infinity,
+      child: ClipRRect(
+        borderRadius: BorderRadius.circular(15.0),
+        child: Image.file(
+          File(
+            widget.mediaPaths[index],
+          ),
+          fit: BoxFit.cover,
+        ),
+      ),
+    );
+  }
```

Comment:
When clicking on the image it shows an error "`Failed assertion: boolean expression must not be null`"